### ISSUE TITLE
Terrain collision doesn't work after square size change

### ISSUE THREAD

**Issue by fr0zi** [GarageGames/Torque3D#1427]
Hello guys.

Lately I was experimenting with terrain sizes. After I change square size to something higher than '1', terrain collisions stop working correctly. It looks like terrain collision mesh isn't recalculated to new terrain size or something. So if I am moving my player towards the terrain he eventually falls off the ground. Below you can find screen with player collision debug turned on:

![torque3d_terrain_bug](https://cloud.githubusercontent.com/assets/606711/10120227/d4d50468-64b1-11e5-9bdc-0c38a27da7b6.jpg)

You can see pink wireframes of collision mesh for player. Player is facing the right side of the screen. As you can see, collision mesh is no longer calculated in front of him, so if I move it further he will fall down.

I checked and this seems to appear in Torque 3.8 RC and 3.7 also.

I'm running Windows 10
ATI Radeon 6850

**Comment by eightyeight** [comment]
Try moving the player further away (10 metres or so). `Player` caches collision information, and only recollects it after having moved a certain distance.

**Comment by fr0zi** [comment]
Yes, I know it works this way. But at this point (where the player is standing) collisions are no longer calculated. If I move player forward (to the right) he will fall down the ground.

![sccopter_debug 2015-09-27 10-23-42-41](https://cloud.githubusercontent.com/assets/606711/10121885/560dd384-6502-11e5-90c6-583b9f3ba62e.jpg)

As you can see above - I moved player forward and collision is not cached. One more step and player will fall.

On the previous screen you can notice a big boulder sphere in the left bottom corner. It indicates the original terrain edge when square size was '1'. Within those borders collisions work fine. But when we move outside those borders, where terrain was resized, collisions works only to some point. 

Like this:
![t3d_terrain_pic](https://cloud.githubusercontent.com/assets/606711/10121922/f2b7128a-6503-11e5-935d-4f2ed1b5c6c1.png)

I hope this will give you some idea :)

**Comment by eightyeight** [comment]
Hmm, interesting. Just to be completely sure, what happens if you use the editor to drag your player like 100 metres into the air and let them fall back onto the terrain? Is the collision fixed?

**Comment by fr0zi** [comment]
I was trying with "Place Player at Camera" and I was moving the camera through whole terrain. When I try to place player anywhere outside this pink rectangle (so outside the orginal terrain size) it falls down.

**Comment by eightyeight** [comment]
Ok, weird!

**Comment by Areloch** [comment]
If you save the mission and reload it, does it still have the incorrect collisions?

**Comment by fr0zi** [comment]
Wow, yes - it works! Collisions work after saving and re-loading the mission.

**Comment by Areloch** [comment]
Alright, awesome!

Sounds like it just doesn't correctly regenerate the collisions during editing. Should be easier to track down then.

**Comment by Lopuska** [comment]
fixed on my code just now. it's the physical mesh transform which is not updated with new matrix after setting changes.
I'll make a PR as soon as possible.
